client/gutenberg/editor/main.jsx

```diff
@@ -32,14 +34,26 @@ class GutenbergEditor extends Component {
 	}
 
 	render() {
-		if ( isEmpty( this.props.siteSlug ) ) {
+		const { postType, siteId, siteSlug } = this.props;
+		if ( isEmpty( siteSlug ) ) {
 			return null;
 		}
 
-		overrideAPIPaths( this.props.siteSlug );
+		overrideAPIPaths( siteSlug );
+
+		const post = { ...mockPost, type: postType };
 
 		return (
-			<Editor settings={ editorSettings } hasFixedToolbar={ true } post={ post } onError={ noop } />
+			<div>
+				<QueryPostTypes siteId={ siteId } />
```

Comment:
@jsnajdr earmarking this for you - `<QueryPostTypes />` relies on the theme slug and two site settings and should refetch when those change. interestingly it doesn't require those dependent inputs.

just noting as it relates to data in general